pos-accounting/src/pool/delta/view_impl.rs

```diff
@@ -97,16 +100,23 @@ impl<P: PoSAccountingView> PoSAccountingView for PoSAccountingDelta<P> {
             self.parent.get_pool_delegations_shares(pool_id).map_err(|_| Error::ViewFail)?;
         let local_shares = self.get_cached_delegations_shares(pool_id);
 
-        match (parent_shares, local_shares) {
-            (None, None) => Ok(None),
-            (None, Some(m)) => Ok(Some(
-                m.into_iter()
-                    .map(signed_to_unsigned_pair)
-                    .collect::<Result<BTreeMap<DelegationId, Amount>, Error>>()?,
-            )),
-            (Some(m), None) => Ok(Some(m)),
-            (Some(m1), Some(m2)) => Ok(Some(sum_maps(m1, m2)?)),
+        let shares = match (parent_shares, local_shares) {
+            (None, None) => return Ok(None),
+            (None, Some(m)) => m
+                .into_iter()
+                .map(signed_to_unsigned_pair)
+                .collect::<Result<BTreeMap<DelegationId, Amount>, Error>>()?,
+            (Some(m), None) => m,
+            (Some(m1), Some(m2)) => sum_maps(m1, m2)?,
+        };
+
+        let mut result = BTreeMap::new();
+        for (id, share) in shares {
+            if self.get_delegation_data(id)?.is_some() {
```

Comment:
Also an explanation here would be nice.

Also, using functional programming would be better, instead of inserts, use iter + filter.